### PR TITLE
Master form warning aab

### PR DESCRIPTION
**Pull request by aab-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/form/form_view.js [hunk 1]
```diff
@@ -148,6 +149,39 @@ var FormView = BasicView.extend({
         }
         return Promise.all(defs);
     },
+    /**
+     * @override
+     */
+    _processArch(arch, fv) {
+        fv.fieldIdsToNames = {}; // maps field ids (identifying <field> nodes) to field names
+        return this._super(...arguments);
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/form/form_view.js:157]: can we use the `...` syntax? I thought it broke something in odoo ?

File: addons/web/static/src/js/views/form/form_view.js [hunk 2]
```diff
@@ -148,6 +149,39 @@ var FormView = BasicView.extend({
...
+        if (node.tag === 'field') {
+            const name = node.attrs.name;
+            let uid = node.attrs.id;
+            if (!uid) {
+                uid = name in fv.fieldIdsToNames ? `${name}__${generateID()}__` : name;
+                node.attrs.id = uid;
+            }
+            fv.fieldIdsToNames[uid] = name;
+        }
+        return this._super(...arguments);
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/form/form_view.js:183]: same?

File: addons/web/static/src/js/views/form/form_renderer.js [hunk 3]
```diff
@@ -1039,16 +1052,9 @@ var FormRenderer = BasicRenderer.extend(WidgetAdapterMixin, {
...
-            var idForLabel = self.idsForLabels[widget.name];
-            // We usually don't support multiple widgets for the same field on the
-            // same view but it is the case with the new settings view on V11.0.
-            // Therefore, we need to retrieve the correct label since it could be
-            // displayed multiple times on the view, otherwise, for example the
-            // enterprise label will be displayed as many times as the field
-            // exists on settings.
-            var $widgets = self.$('.o_field_widget[name=' + widget.name + ']');
+            const fieldId = self.widgetsToFieldIds.get(widget);
+            const idForLabel = self.idsForLabels[fieldId];
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/form/form_renderer.js:1056]: yay

File: addons/web/static/src/js/views/form/form_renderer.js [hunk 4]
```diff
@@ -120,12 +124,16 @@ var FormRenderer = BasicRenderer.extend(WidgetAdapterMixin, {
         $labels.removeClass('o_field_invalid');
 
         _.each(fieldNames, function (fieldName) {
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/form/form_renderer.js:126]: (stylistic note)I would write the next few lines like this:
```js
const widgetList = this.allFieldWidgets[this.state.id];
for (let fieldName of fieldNames) {
  for (let widget of widgetList.filter(w => w.name === fieldName) {
    ...
  }
}
```

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo rebase-ff

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**